zcash_primitives/src/transaction/mod.rs

```diff
@@ -881,6 +1074,24 @@ impl Transaction {
         sapling_serialization::write_v5_bundle(writer, self.sapling_bundle.as_ref())
     }
 
+    #[cfg(zcash_unstable = "nu6")] /* TODO nu7 */
+    pub fn write_v7<W: Write>(&self, mut writer: W) -> io::Result<()> {
+        if self.sprout_bundle.is_some() {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "Sprout components cannot be present when serializing to the V7 transaction format.",
+            ));
+        }
+        self.write_v5_header(&mut writer)?;
+        self.write_transparent(&mut writer)?;
+        self.write_v5_sapling(&mut writer)?;
```

Comment:
rename to `write_sapling()`

Reply:
same here?

Reply:
well yes, the functionality is now extended to v6